flexget/plugins/input/anidb_feed.py

```diff
@@ -0,0 +1,479 @@
+            raise plugin.PluginError('Unable to reach anidb feed url: %s' % url)
+
+        fixed_xml = self.make_feedparser_friendly(r.content)
+        try:
+            if fixed_xml:
+                parsed_xml = feedparser.parse(fixed_xml)
+            else:
+                parsed_xml = feedparser.parse(r.content)
+            xml_feed = parsed_xml.feed
+            if 'error' in xml_feed:
+                if 'code' in xml_feed['error']:
+                    if 'description' in xml_feed['error']:
+                        log.error(
+                            'Error code: %s detail: %s' % (xml_feed['error']['code'], xml_feed['error']['description']))
+                    else:
+                        log.error('Error code: %s' % xml_feed['error']['code'])
+        except Exception as ex:
+            log.error('Unable to parse the XML from url: %s error: %s' % (url, ex))
+            return []
+
+        if not len(parsed_xml.entries) > 0:
+            log.info('No entries returned from xml.')
+            return []
+
+        entries = self.parse_from_xml(parsed_xml.entries, config)
+        if len(entries) == 0:
+            log.verbose('No entries parsed from xml.')
+        # else:
+        #   dump_entry(entries[0])
+
+        return entries
+
+    def make_feedparser_friendly(self, data):
+        try:
+            dom = minidom.parseString(data)
+            entries = dom.getElementsByTagName('entry')
+            for entry in entries:
+                items_ns = entry.getElementsByTagNameNS(NAMESPACE_URL, NAMESPACE_TAGNAME)
+                if items_ns:
+                    for node_ns in items_ns:
+                        dt_nodes = node_ns.getElementsByTagName(NAMESPACE_NAME + ':dt')
+                        dd_nodes = node_ns.getElementsByTagName(NAMESPACE_NAME + ':dd')
+                        if len(dt_nodes) != len(dd_nodes):
+                            logging.warning('unexpected nodecount')
+                            continue
+                        for idx, dt_node in enumerate(dt_nodes):
+                            if len(dt_node.childNodes) >= 1 and len(dd_nodes[idx].childNodes) >= 1:
+                                if dt_node.childNodes[0].nodeValue is not None and dd_nodes[idx].childNodes[0].nodeValue is not None:
+                                    dt_node.tagName = NAMESPACE_NAME + ':%s' % dt_node.childNodes[0].nodeValue.replace(' ', '_')
+                                    dt_node.setAttribute('name', dt_node.childNodes[0].nodeValue)
+                                    dt_node.setAttribute('value', dd_nodes[idx].childNodes[0].nodeValue)
+                                    dt_node.removeChild(dt_node.childNodes[0])
+                                    node_ns.removeChild(dd_nodes[idx])
+                                    entry.appendChild(dt_node)
+        except Exception as ex:
+            logging.debug('Unable to rename nodes in XML: %s' % ex)
+            return None
```

Comment:
> Way too wide exception catch that completely hides crashes.
Yes i still have problems understanding how to handle errors in flexget/python, can you give me an example on how to handle this case?
I use the empty/None returns on errors that i won't handle, so should i just write the code as if no error might happen and put a try on the topmost caller? Should i expect the caller of my plugin to have a try/catch? I mean in this specific case i can either throw an exception and flexget just stops/crashes or i can just do a error log and use the designed alternate code-path?

I don't fully understand the issue with none/empty objects, if a caller ask for a object and the function cant get a valid object, i'm supposed to throw a exception? If that's the case, sure i can change those returns to raise specific exceptions instead?

PS: Unfortunately my c++ background is not helping either, since the projects i was involved in the past not even had exception handling enabled. So i'm used to try always return a valid object, even if its a empty object and as fallback return a none object that the caller has to handles. In the past i noticed that no one really handles specific exceptions anyway and just lets the application crash/stop, which is also bad design. I really liked code contracts in c#, so exceptions handling is not my strong side.